### PR TITLE
[@types/puppeteer] add return type 'ElementHandle' to page.evaluateHandle

### PR DESCRIPTION
**Pull request by dreamwayjgs**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <[page.evaluateHandle](https://pptr.dev/#?product=Puppeteer&version=main&show=api-pageevaluatehandlepagefunction-args)>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/puppeteer/index.d.ts
Comment (cameronhunter) [inline types/puppeteer/index.d.ts:287]: I'm surprised that we need these `tslint:disable`s – do you know why it thinks they are unnecessary?

Comment (dreamwayjgs) [reply types/puppeteer/index.d.ts:287]: I'm not confident either, but puppeteer's doc said:
```
This function will return a JSHandle by default, however if your pageFunction returns an HTML element you will get back an ElementHandle:
```
When I first met this situation, I thought when using <code>evaluateHandle</code> in typescript would become like this:
```
const elementHandle = await page.evaluateHandle<HTMLAnchorElement>( fn ) // and fn returns HTMLAnchorElement
```
But, puppeteer [source comment](https://github.com/puppeteer/puppeteer/blob/444c7e0e4729748d4c124bf8b4948a3d74dcf726/src/common/Page.ts#L768:) said we should use:
```
const button = await page.evaluateHandle<ElementHandle>(...);
```
So, I tried [same type definition](https://github.com/puppeteer/puppeteer/blob/444c7e0e4729748d4c124bf8b4948a3d74dcf726/src/common/Page.ts#L774) to that code but, the <code>HandlerType</code> generic used only return type and cause <code>no-unnecessary-generics</code>. <code>HandlerType</code> depends on <code>pageFunction</code>'s return type but it doesn't have any relation with <code>JSHandle</code> or <code>ElementHandle</code> type(or just I failed to make fancier solution).

I think this issue that discussed in https://github.com/Microsoft/dtslint/issues/76. So, I tried the solution in the link https://github.com/Microsoft/dtslint/issues/76#issuecomment-364197479

**Comment by sandersn** [comment]
@cameronhunter did @dreamwayjgs answer about tslint exceptions make sense? puppeteer usage is so widespread that I'm not going to merge this without signoff from a maintainer.

File: types/puppeteer/index.d.ts
Comment (amcasey) [reply types/puppeteer/index.d.ts:287]: While I have to admit I'm curious why these types are here when puppeteer also ships a copy, it seems like matching their version is the way to go:
```ts
evaluateHandle<HandlerType extends JSHandle = JSHandle>(pageFunction: EvaluateHandleFn, ...args: SerializableOrJSHandle[]): Promise<HandlerType>;
```

Comment (amcasey) [reply types/puppeteer/index.d.ts:287]: (i.e. I'm okay with suppressing the lint warning)

File: types/puppeteer/index.d.ts
Comment (amcasey) [inline types/puppeteer/index.d.ts:288]: This conditional type doesn't appear to be present in the original.  Looks like it's just supposed to return `Promise<HandlerType>`?

**Review by amcasey — CHANGES_REQUESTED** [review]
Without additional context explaining why these types should differ from the types in the underlying library, I'd suggest matching them exactly.

File: types/puppeteer/index.d.ts
Comment (dreamwayjgs) [reply types/puppeteer/index.d.ts:288]: You're right. It looks like my mistake. I thought I had to check if the type was <code>ElementHandle</code> or <code>JSHandle</code>.  In hindsight, there was no need to do that.
Either way, the tslint exception is needed to pass the dtslint test. Should I fix and reopen PR?

Comment (dreamwayjgs) [reply types/puppeteer/index.d.ts]: I've committed changes. Thank you.

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by dreamwayjgs** [comment]
Ready to merge

**Pull request merged**
